File: DroidsAPI/src/main/java/com/github/rochedo098/droidsapi/api/events/MemberEventHandler.java
package com.github.rochedo098.droidsapi.api.events;

import com.github.rochedo098.droidsapi.api.types.MemberType;
import net.minecraft.entity.LivingEntity;
import org.jetbrains.annotations.ApiStatus;

import java.util.ArrayList;
import java.util.List;

@ApiStatus.Experimental
public class MemberEventHandler {
    private final List<MemberEventListener> listeners = new ArrayList<>();

    public void addListener(MemberEventListener toAdd) {
        listeners.add(toAdd);
    }

    @ApiStatus.Internal
    public void runChangeEvent(LivingEntity entity, MemberType member) {
        for (MemberEventListener ml : listeners) {
            ml.onChange(entity, member);
        }
    }
}

File: src/main/kotlin/com/github/rochedo098/droids/api/events/MemberEventHandler.kt
package com.github.rochedo098.droids.api.events

import com.github.rochedo098.droids.api.types.MemberType
import net.minecraft.entity.LivingEntity

class MemberEventHandler {
    private var listeners: MutableList<MemberEventListener> = ArrayList()

    fun addListener(toAdd: MemberEventListener) { listeners.add(toAdd) }

    fun runChangeEvent(entity: LivingEntity, member: MemberType) {
        for (ls in listeners) {
            ls.onChange(entity, member)
        }
    }
}